File: test-suite-kotlin2-ksp2/src/test/kotlin/io/micronaut/docs/ioc/mappers/ContactMappers.kt
package io.micronaut.docs.ioc.mappers

//tag::imports[]
import io.micronaut.context.annotation.Mapper
//end::imports[]
import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "SimpleMapperSpec")
//tag::class[]
interface ContactMappers {
    @Mapper
    fun toEntity(contactForm: ContactForm) : ContactEntity
}
//end::class[]

File: test-suite-python/src/test/python/micronaut/docs/ioc/mappers/ContactMappers.py
from abc import ABC, abstractmethod

from .ContactEntity import ContactEntity
from .ContactForm import ContactForm

# tag::imports[]
from micronaut.context.annotation import Mapper
# end::imports[]


# tag::class[]
class ContactMappers(ABC):
    @Mapper
    @abstractmethod
    def to_entity(self, contact_form: ContactForm) -> ContactEntity:
        pass
# end::class[]
